src/components/Plan.js

```diff
@@ -18,80 +18,114 @@ import React, { Component } from 'react';
 
 import TokenInput from './TokenInput';
 
-class Plan extends Component {
+class IssueTree extends Component {
 
     render() {
-        let phases = [...new Set(
-            [].concat(...this.props.issues.map(issue =>
-                issue.labels.map(label => label.name)
-            )))]
-            .filter(label => label.startsWith("phase:"))
-            .sort((a, b) => {
-                return Number(a.split(":")[1]) -
-                    Number(b.split(":")[1]);
-            });
+        let fields = this.props.fields;
+        let items = this.props.items;
+        let renderItem = this.props.renderItem;
 
-        let phasedIssues = {};
-        phases.forEach(phase => {
-            phasedIssues[phase] = this.props.issues.filter(issue => {
-                return issue.labels.map(label => label.name).includes(phase);
-            });
-        });
+        if (fields.length === 0) {
+            return items.sort(this.props.sortItems).map(item => renderItem(item));
+        }
+
+        let thisLevel = fields[0];
+
+        let thisField = thisLevel.field;
+        let unbucketedName = thisLevel.unbucketedName || 'noname';
+
+        let headings = [...new Set(items.filter(thisField).map(thisField))].sort(thisLevel.sort);
 
-        // Add unphased issues as an extra section
-        phasedIssues["unphased"] = this.props.issues.filter(issue => {
-            const labels = issue.labels.map(label => label.name);
-            return !labels.some(label => label.startsWith("phase:"));
+        let buckets = {};
+        headings.forEach(heading => {
+            buckets[heading] = items.filter(item => thisField(item) === heading);
         });
-        if (phasedIssues["unphased"].length > 0) {
-            phases.push("unphased");
+
+        // Put any of the items that didn't land in a headered bucket into the
+        // 'unbucketed' category.
+        let unbucketed = items.filter(item =>
+            !Object.values(buckets).reduce(Array.concat, []).includes(item));
+
+        if (unbucketed.length > 0) {
```

Comment:
Maybe only create unbucketed when `unbucketedName` is supplied and remove the default unbucketed name?

Reply:
Agreed; that sounds more flexible.